skyportal/handlers/api/comment.py

```diff
@@ -59,46 +75,77 @@ def get(self, comment_id, associated_resource_type=None):
                 schema: Error
         """
 
-        if associated_resource_type is None:
-            associated_resource_type = 'object'
+        try:
+            comment_id = int(comment_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid (scalar integer) comment ID. ")
+
+        # the default is to comment on an object
+        if associated_resource_type.lower() == "sources":
+            try:
+                comment = Comment.get_if_accessible_by(
+                    comment_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error('Could not find any accessible comments.')
+            comment_resource_id_str = str(comment.obj_id)
 
-        if associated_resource_type.lower() == "object":  # comment on object (default)
-            comment = Comment.get_if_accessible_by(
-                comment_id, self.current_user, raise_if_none=True
-            )
         elif associated_resource_type.lower() == "spectrum":
-            comment = CommentOnSpectrum.get_if_accessible_by(
-                comment_id, self.current_user, raise_if_none=True
-            )
+            try:
+                comment = CommentOnSpectrum.get_if_accessible_by(
+                    comment_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error('Could not find any accessible comments.')
+            comment_resource_id_str = str(comment.spectrum_id)
+
         # add more options using elif
         else:
             return self.error(
                 f'Unsupported input "{associated_resource_type}" given as "associated_resource_type" argument.'
             )
 
+        if comment_resource_id_str != resource_id:
+            return self.error(
+                f'Comment resource ID does not match resource ID given in path ({resource_id})'
+            )
         return self.success(data=comment)
 
     @permissions(['Comment'])
-    def post(self):
+    def post(self, associated_resource_type, resource_id):
         """
         ---
         description: Post a comment
         tags:
           - comments
+        parameters:
+          - in: path
+            name: associated_resource_type
+            required: true
+            schema:
+              type: string
```

Comment:
This should be an enum (same elsewhere)

Reply:
I'm not sure how to do this as an ENUM. Can you think of an example already using that so I can have a look?

Reply:
Grep for `enum` in `api/source.py`

Reply:
OK, so it isn't enforced as a data type, just documented as a string that has a closed set of options.